astropy/time/formats.py

```diff
@@ -1405,6 +1551,28 @@ class TimeYearDayTime(TimeISO):
                 '%Y:%j',
                 '{year:d}:{yday:03d}'))
 
+    # Define positions and starting delimiter for year, month, day, hour,
+    # minute, seconds components of an ISO time. This is used by the fast
+    # C-parser parse_ymdhms_times()
+    #
+    #  "2000:123:13:14:15.678"
+    #   012345678901234567890
+    #   yyyy:ddd:hh:mm:ss.fff
+    # Parsed as ('yyyy', ':ddd', ':hh', ':mm', ':ss', '.fff')
+    #
+    # delims: character at corresponding `starts` position (0 => no character)
+    # starts: position where component starts (including delimiter if present)
+    # stops: position where component ends (-1 => continue to end of string)
+
+    # Before: yr mon  doy     hour      minute    second    frac
+    use_fast_parser = True
```

Comment:
Would it be an idea to abstract this out a bit more, so that the class perhaps defines the parser proper instead of the pieces needed for it? I think it might be a bit more future proof. I.e., something like
```
fast_parser = CParserWrapper(delims=..., starts=..., stops=..., ...)
```
and then the `CParserWrapper.__call__` method would have most of what is now in  `set_jds_fast`.

Reply:
Another option (closer to what you have) would be to here define `fast_parser_pars = dict(delims=..., starts=..., ...)`.

But I think I'd prefer the option of a user-defined parser.

Though I can also see the argument that really the `TimeFormat` class *is* the parser so this just belongs here...

Reply:
I see the benefit of encapsulating the parameters in a single attribute `fast_parser_pars`. The downside is that you cannot easily use attribute inheritance ala `TimeISO` and `TimeISOT`. But repeating those pars in `TimeISOT` is not the end of the world, so I'll plan on that new single container.

About abstracting this out, I've become much more resistant to making abstractions to solve a problem that we don't have yet. This functionality can be fully customized by a subclass that overrides `set_jds_fast`. This seems simple and gives subclasses the ability to do whatever is needed if they invent a different fast parser.

Reply:
OK, makes sense not to abstract out. One thought to make it easier to extend in future if there ever is a need: how about if the python and fast versions not not set jd1, jd2 themselves but rather return jd1, jd2 and those are then stored on the class inside `set_jds`?

Reply:
> not set jd1, jd2 themselves but rather return jd1, jd2

Done.